bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/EventFilterHandler.java

```diff
@@ -0,0 +1,362 @@
+        final ThingBuilder thingBuilder = editThing();
+        if (config.maxEvents == null || config.maxEvents.compareTo(new BigDecimal(0)) < 1) {
+            thingBuilder.withoutChannels(currentChannels);
+            updateThing(thingBuilder.build());
+            return;
+        }
+        generateExpectedChannelList(config.maxEvents.intValue());
+
+        currentChannels.stream().filter((Channel current) -> {
+            @Nullable
+            String currentGroupId = current.getUID().getGroupId();
+            if (currentGroupId == null) {
+                return true;
+            }
+            for (ResultChannelSet channelSet : resultChannels) {
+                if (channelSet.resultGroup.getId().contentEquals(currentGroupId)) {
+                    return false;
+                }
+            }
+            return true;
+        }).forEach((Channel toDelete) -> {
+            thingBuilder.withoutChannel(toDelete.getUID());
+        });
+
+        resultChannels.stream().filter((ResultChannelSet current) -> {
+            return (getThing().getChannelsOfGroup(current.resultGroup.toString()).size() == 0);
+        }).forEach((ResultChannelSet current) -> {
+            for (ChannelBuilder builder : handlerCallback.createChannelBuilders(current.resultGroup, GROUP_TYPE_UID)) {
+                Channel currentChannel = builder.build();
+                @Nullable
+                Channel existingChannel = getThing().getChannel(currentChannel.getUID());
+                if (existingChannel == null) {
+                    thingBuilder.withChannel(currentChannel);
+                }
+            }
+        });
+
+        updateThing(thingBuilder.build());
+    }
+
+    /**
+     * Updates all states and channels.
+     */
+    private void updateStates() {
+        final Bridge iCalendarBridge = getBridge();
+        if (iCalendarBridge == null) {
+            logger.debug("Bridge not instantiated!");
+            return;
+        }
+        final ICalendarHandler iCalendarHandler = (ICalendarHandler) iCalendarBridge.getHandler();
+        if (iCalendarHandler == null) {
+            logger.debug("ICalendarHandler not instantiated!");
+            return;
+        }
+        final EventFilterConfiguration config = configuration;
+        if (config == null) {
+            logger.debug("Configuration not instantiated!");
+            return;
+        }
+        @Nullable
+        final AbstractPresentableCalendar cal = iCalendarHandler.getRuntimeCalendar();
+        if (cal != null) {
+            updateStatus(ThingStatus.ONLINE);
+
+            String textFilterValue = config.textEventValue;
+            EventTextFilter filter = null;
+            if (textFilterValue != null) {
+                try {
+                    if (config.textEventField == null || config.textValueType == null) {
+                        throw new IllegalArgumentException("Config is broken.");
```

Comment:
`IllegalArgumentException` is an unchecked exception, which are not meant to be cought. Is an exception needed at all here? Could you simply return and set the ThingStatus accordingly?

Reply:
I just did not want to reinvent the wheel and am catching the exception. Do you know a better one or should i create a own exception class for broken configs?

Reply:
Creating a custom exception would be a valid approach. But I'm wondering if an exception is needed at all here. Can't you simply return like in line 265? BTW you should set the ThingStatus accordingly in this line.

Reply:
OK